File: backend/src/main/java/com/grouply/backend/user/Dtos/UserDTO.java
package com.grouply.backend.user.Dtos;

import com.grouply.backend.user.Role;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {

    private Long id;
    private String firstName;
    private String lastName;
    private String username;
    private String email;
    private String avatarUrl;
    private Role role;
    private LocalDateTime createdAt;
}

File: frontend/src/dtos/models_dtos/user_dto.ts
import type { Role } from "../enums/Role"


export class UserDTO {
    id: number
    firstName: string
    lastName: string
    username: string
    email: string
    avatarUrl?: string
    role: Role 

    constructor(id:number, firstName: string,lastName: string,username: string,email: string,
    role: Role ,avatarUrl?: string ) {
        this.id = id;
        this.firstName = firstName
        this.lastName = lastName
        this.email = email
        this.username = username
        this.avatarUrl = avatarUrl;
        this.role = role
    }
    
}